.pre-commit-config.yaml

```diff
@@ -138,3 +138,11 @@ repos:
       - id: renovate-config-validator
         args:
           - --strict
+
+  - repo: local
+    hooks:
+      - id: codeowners-validator
+        name: CODEOWNERS Validator
+        entry: ./codeowners-validator.sh
+        language: script
+        files: ^CODEOWNERS$
```

Comment:
We should add this to the existing group of `repo: local` hooks.

Additionally, we don't need to use a shell script and a custom docker file to build from. We can just use the image from the project itself: https://github.com/mszostok/codeowners-validator/pkgs/container/codeowners-validator

For that we can use the `docker_image` pre-commit language:

```yaml
- id: codeowners-validator
  name: CODEOWNERS validator
  language: docker_image
  pass_filenames: false
  files: CODEOWNERS$
  entry: >
    -e REPOSITORY_PATH="."
    -e CHECKS="files,duppatterns,syntax"
    ghcr.io/mszostok/codeowners-validator:v0.7.4
```

We can leave out the "owners" check. I don't think it's much of a concern, and the original issue is only trying to fix the issue where files move around and are no longer included in the codeowners. If we want to include it, that's fine, and it should work because the `GITHUB_ACCESS_TOKEN` is not _required_, and all our codeowner teams are public. It would only cause an issue if someone hit the rate limit locally. In CI we can use it from the environment variable like we do other things... but I really don't think the `owners` check is a real concern for us and we can ignore the access token issue altogether.

My only outstanding question is whether we should enable the two experimental checks, notowned, which ensures there are no "unowned" files, and "avoid-shadowing", which helps make the codeowners file easier to reason about. I don't think they're necessary for this PR though. Getting the basic checks in would be better than delaying this to decide about the experimental ones.